From 4c7507b00977b1d059b5451e3ac9eb16c8a9849c Mon Sep 17 00:00:00 2001
From: CI Assistant <chdauto@intel.com>
Date: Fri, 20 Dec 2019 14:14:54 +0200
Subject: [PATCH] Merge pull request #1447 in SW_UGW/linux from
 feature/UGW_SW-45101-map-e-porting-from-7.x-to-8.x_841 to 8.4.1

* commit '266d51792917f1280b50f4018966b4f7a0a19004':
  GW_SW-45101: Porting MAP-E acceleration to 8x
---
 include/net/ppa/ppa_api.h      |  1 +
 include/net/ppa/ppa_stack_al.h | 19 +++++++++++++++++++
 2 files changed, 20 insertions(+)

diff --git a/include/net/ppa/ppa_api.h b/include/net/ppa/ppa_api.h
index 78f2beb0cdd4..ebc0ac299ddd 100644
--- a/include/net/ppa/ppa_api.h
+++ b/include/net/ppa/ppa_api.h
@@ -192,6 +192,7 @@ typedef enum {
 #endif
 	QOS_METER,
 	WMM_QOS_DSCP_MAP,
+	TUNNEL_MAP_E,
 	MAX_CAPS
 } PPA_API_CAPS;
 
diff --git a/include/net/ppa/ppa_stack_al.h b/include/net/ppa/ppa_stack_al.h
index ae4a90367f4c..588ccd2ea873 100644
--- a/include/net/ppa/ppa_stack_al.h
+++ b/include/net/ppa/ppa_stack_al.h
@@ -1661,6 +1661,25 @@ int32_t ppa_form_gre_hdr(PPA_NETIF *dev,
 		uint16_t dataLen,
 		uint8_t *pHdr,
 		uint16_t *len);
+/*! \brief  This function forms IPv6 header (DSLite) for given interface.
+  \param[out]  dev  Pointer to the buffer to copy header.
+  \param[in]   pointer to netdev struct.
+  \param[in]   payload length.
+  \return    Returns PPA_SUCCESS value if header is formed.
+  \note
+ */
+int32_t ppa_get_ipv6_tnl_iph(struct ipv6hdr* ip6h,
+   struct net_device *dev,
+   uint16_t dataLen);
+/*! \brief  This function detects mesh mode for MapE session.
+  \param[in]   IP destination IPv4  address of the map-e session.
+  \param[in]   net dev pointer for map-e interface.
+  \return    Returns true if mesh mode is detected else false.
+  \note
+ */
+bool ppa_is_mape_mesh_session(uint32_t dest_ip,
+                struct net_device *dev);
+
 /*! \brief This function returns EOGRE inner destination mac address.
 	\param[out] mac Pointer to destination mac address.
 	\param[in] skb Pointer to the packet buffer.
